The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8982395525/job/24669929440) [(plain)](https://github.com/rust-lang/rust/commit/bbe7f05819fe9168537a29aac5a4477121390e56/checks/24669929440/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:0ad4b8fadaa221de15dcec353f45205ec38ea70b)
Download action repository 'actions/upload-artifact@v4' (SHA:65462800fd760344b1a7b4382951275a0abb4808)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.412   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.424 Collecting boolean-py==4.0
#13 0.427   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.439 Collecting chardet==5.1.0
---
#13 3.455 Building wheels for collected packages: reuse
#13 3.456   Building wheel for reuse (pyproject.toml): started
#13 3.784   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.785   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.785   Stored in directory: /tmp/pip-ephem-wheel-cache-dmbzymmz/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.788 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.815   Attempting uninstall: setuptools
#13 3.815     Found existing installation: setuptools 59.6.0
#13 3.816     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 3.816     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 3.817     Can't uninstall 'setuptools'. No files were found to uninstall.
#13 4.491 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#13 4.492 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 5.007 Collecting virtualenv
#13 5.054   Downloading virtualenv-20.26.1-py3-none-any.whl (3.9 MB)
#13 5.109      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.9/3.9 MB 74.3 MB/s eta 0:00:00
#13 5.145 Collecting distlib<1,>=0.3.7
#13 5.149   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.185 Collecting filelock<4,>=3.12.2
#13 5.188   Downloading filelock-3.14.0-py3-none-any.whl (12 kB)
#13 5.188   Downloading filelock-3.14.0-py3-none-any.whl (12 kB)
#13 5.216 Collecting platformdirs<5,>=3.9.1
#13 5.220   Downloading platformdirs-4.2.1-py3-none-any.whl (17 kB)
#13 5.308 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.476 Successfully installed distlib-0.3.8 filelock-3.14.0 platformdirs-4.2.1 virtualenv-20.26.1
#13 DONE 5.6s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      194496 kB
DirectMap2M:     7145472 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/e2865db2b990259e7873393f31e181a53dbc0bfc/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-e2865db2b990259e7873393f31e181a53dbc0bfc-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 25.06s
##[endgroup]
fmt check
##[error]Diff in /checkout/tests/run-make/rustdoc-map-file/rmake.rs at line 10:
         .output(&out_dir)
         .run();
     assert!(
-        Command::new("python").arg("validate_json.py").arg(&out_dir).status().unwrap().success());
+        Command::new("python").arg("validate_json.py").arg(&out_dir).status().unwrap().success()
 }
 
 
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/tests/run-make/stdin-rustc/rmake.rs" "/checkout/tests/run-make/core-no-fp-fmt-parse/rmake.rs" "/checkout/tests/run-make/wasm-spurious-import/rmake.rs" "/checkout/tests/run-make/repr128-dwarf/rmake.rs" "/checkout/tests/run-make/exit-code/rmake.rs" "/checkout/tests/run-make/wasm-stringify-ints-small/rmake.rs" "/checkout/tests/run-make/wasm-abi/rmake.rs" "/checkout/tests/run-make/rust-lld/rmake.rs" "/checkout/tests/run-make/rustdoc-map-file/rmake.rs" "/checkout/tests/run-make/wasm-symbols-not-exported/rmake.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Tue May  7 08:50:34 UTC 2024
  network time: Tue, 07 May 2024 08:50:34 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/tools/tidy/src/allowed_run_make_makefiles.txt
+++ b/src/tools/tidy/src/allowed_run_make_makefiles.txt
@@ -245,7 +245,6 @@ run-make/rlib-format-packed-bundled-libs/Makefile
 run-make/rmeta-preferred/Makefile
 run-make/rustc-macro-dep-files/Makefile
 run-make/rustdoc-io-error/Makefile
-run-make/rustdoc-map-file/Makefile
 run-make/rustdoc-output-path/Makefile
 run-make/rustdoc-scrape-examples-invalid-expr/Makefile
 run-make/rustdoc-scrape-examples-macros/Makefile

--- a/tests/run-make/rustdoc-map-file/Makefile
+++ b/tests/run-make/rustdoc-map-file/Makefile
@@ -1,5 +1,0 @@
-include ../tools.mk
-
-all:
-	$(RUSTDOC) -Z unstable-options --generate-redirect-map foo.rs -o "$(TMPDIR)/out"
-	"$(PYTHON)" validate_json.py "$(TMPDIR)/out"

--- a/tests/run-make/rustdoc-map-file/rmake.rs
+++ b/tests/run-make/rustdoc-map-file/rmake.rs
@@ -1,0 +1,15 @@
+use run_make_support::{rustdoc, tmp_dir};
+use std::process::Command;
+
+fn main() {
+    let out_dir = tmp_dir().join("out");
+    rustdoc()
+        .input("foo.rs")
+        .arg("-Zunstable-options")
+        .arg("--generate-redirect-map")
+        .output(&out_dir)
+        .run();
+    // FIXME (GuillaumeGomez): Port the python script to Rust as well.
+    let python = std::env::var("PYTHON").unwrap_or("python".into());
+    assert!(Command::new(python).arg("validate_json.py").arg(&out_dir).status().unwrap().success());
+}

